### DISCUSSION TITLE
Is there a more convenient way to use the `include note` feature?

### DISCUSSION THREAD

**Discussion by shutaozhenzhen** [TriliumNext/Trilium#3030] [Q&A] [answered]
Is it possible to set `include note` to automatically form when copying or dragging `note` to the editing page? Or provide a shortcut evoking method like `[[]]`?
Or, how to insert `include note` via JavaScript? What is the ckeditor command?

**Comment by shutaozhenzhen** [comment]
anyone has any idea?

**Reply by sigaloid** [reply]
It would be better to file a specific feature request to add that feature, to the [keyboard shortcuts](https://github.com/zadam/trilium/wiki/Keyboard-shortcuts): https://github.com/zadam/trilium/issues/new?assignees=&labels=Type%3A+Enhancement&template=feature_request.yml&title=%28Feature+request%29+

**Comment by salmund** [comment]
[https://github.com/salmund/trilium_better_include/tree/main](url)

**Comment by salmund** [comment, marked as answer]
If you want to include a note in trilium using javascript it's pretty simple (but it's not explained in the API) :

First, you must get a CKEditor instance of your texte note :
```
const editor = await api.getActiveContextTextEditor()
```

Then you can use this code to insert the box automatically. Obviously, you have to determine the id of the note to be inserted beforehand, but it is quite easy.

```
 editor.model.change(writer => {
                const includedNote = writer.createElement('includeNote', {
                noteId: noteToInsertId,
                boxSize: "medium"
            })
            editor.model.insertContent(includedNote);
            })
            }
```
